src/linear_solver/NNLS_solver.cpp

```diff
@@ -365,8 +376,57 @@ bool NNLS_solver::solve(){
         moveToActiveSet(infeasibleIdx);
       }
     }
-    
   }
 }
 
+Epetra_CrsMatrix NNLS_solver::allocateMatrixSingleCore(const Epetra_CrsMatrix &A){
```

Comment:
This function is returning a matrix which will then be copied into another matrix. Consider passing in the output matrix as a reference and then updating it within the function it to improve efficiency.

Reply:
As discussed with @tofstie , since this function is used to initialize matrices in the constructor, it's better for it to return an Epetra matrix. Marking this comment as resolved.